tests/lean/run/2846.lean

```diff
@@ -30,31 +30,41 @@ def Nat.pos_pow_of_pos' {n : Nat} (m : Nat) : 0 < n → 0 < n ^ m := Nat.pos_pow
 
 /-!
 Repetition of a named argument, only the first is printed as a named argument.
+The second is made hygienic.
 -/
 def foo (n n : Nat) : Fin (n + 1) := 0
 
-/-- info: foo (n : Nat) : (n : Nat) → Fin (n + 1) -/
+/-- info: foo (n : Nat) : (n✝ : Nat) → Fin (n✝ + 1) -/
```

Comment:
@david-christiansen What do you think about this refinement to #2846? It uses daggered names here because we want to know exactly which arguments are the ones that can be used with named arguments.

The rule is a bit wonky. It still has that the first hygienic name causes a shift to after the colon, but if it's a dependent arrow you see the hygienic name. Potentially we could make dependent arrows eagerly have their arguments come before the colon — maybe that refinement can come in a future PR?

In any case, your example of `String.append : String → String → String` still comes out as such.

Reply:
Modified the rule like I mentioned. Now this comes out as 
```
foo (n n✝ : Nat) : Fin (n✝ + 1)
```
We can't avoid the hygienic name because the type depends on `n✝`, and we can't give it a different name because it's not useable for named arguments.

Reply:
By the way, with this version it should now be possible to give a reproducer for the issue where multiple dagger names run together because "you don't need a `ppSpace` between `ident` because spaces are automatically inserted", which is false in the presence of dagger names.

Reply:
@digama0 That's fixed in this PR :-) Your test case from another issue made it in to the tests too.

Reply:
I think that both proposed ways of showing the signature (before and after the colon) are reasonable here, for the reason you say :)